### PR TITLE
Temporary optimizer support for ort format models in non minimal build

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request:

- Checks the input optimizer model to determine if it is an ort format model before trying to parse it.
- fixes the C/C++ API documentation GH actions.

### CODE REVIEW

File: orttraining/orttraining/training_api/optimizer.cc [hunk 1]
```diff
@@ -63,19 +64,25 @@ std::unique_ptr<OptimizerAlgorithmBase> OptimizerAlorithmFactory::CreateInstance
...
-                          logging::LoggingManager::DefaultLogger())
-                  .IsOK());
-  Graph& graph = model->MainGraph();
-  for (auto& node : graph.Nodes()) {
-    if (node.Domain() == kMSDomain && (node.OpType() == "AdamWOptimizer" || node.OpType() == "SGDOptimizerV2")) {
-      auto domain_type_pair = std::make_pair(node.Domain(), node.OpType());
-      if (opt_type_to_freq_map.find(domain_type_pair) == opt_type_to_freq_map.end()) {
-        opt_type_to_freq_map[domain_type_pair] = 0;
-      }
+  if (fbs::utils::IsOrtFormatModel(ToWideString(optim_path_or_bytes))) {
```
Comment (edgchen1) [inline orttraining/orttraining/training_api/optimizer.cc:67]: we can call `ToWideString()` (or preferably the more descriptive `ToPathString()`) just once:
```c++
if (const auto optim_path_str = ToPathString(optim_path_or_bytes);
    fbs::utils::IsOrtFormatModel(optim_path_str)) {
  ...
} else {
  ...
}
```

also `optim_path_or_bytes` implies that we would handle model bytes here too, but that doesn't look like the case.

File: include/onnxruntime/core/session/onnxruntime_cxx_api.h [hunk 2]
```diff
@@ -1382,7 +1382,7 @@ struct Value : detail::ValueImpl<OrtValue> {
    * \tparam T - the type of the value.
    * \param domain - zero terminated utf-8 string. Domain of the type.
    * \param type_name - zero terminated utf-8 string. Name of the type.
-   * \param value - the value to be wrapped.
+   * \return value - the value to be wrapped.
    */
   template <typename T>
   static Value CreateOpaque(const char* domain, const char* type_name, const T&);  ///< Wraps OrtApi::CreateOpaqueValue
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply include/onnxruntime/core/session/onnxruntime_cxx_api.h:1388]: Oh, I did not read the argument documentation. Thanks for catching this.

File: orttraining/orttraining/training_api/optimizer.cc [hunk 3]
```diff
@@ -60,29 +61,36 @@ Status GraphInputsAreExpected(gsl::span<std::string> actual_graph_inputs,
...
+  if (const auto optim_path_str = ToPathString(optim_path);
+      fbs::utils::IsOrtFormatModel(optim_path_str)) {
+    // TODO (baijumeswani): Figure out the best way to extract the optimizer type
+    //                      from an ort format model.
+    opt_type_to_freq_map[std::make_pair(kMSDomain, "AdamWOptimizer")] = 1;
+  } else {
+    std::shared_ptr<Model> model;
+    ORT_ENFORCE(Model::Load(optim_path_str, model, nullptr,
+                            logging::LoggingManager::DefaultLogger())
+                    .IsOK());
```
Comment (pengwa) [inline orttraining/orttraining/training_api/optimizer.cc:76]: do you still need this logic, will offline model still generate protobuf?

**Review by pengwa — APPROVED** [review]

File: orttraining/orttraining/training_api/optimizer.cc (hunk 3 above)
Comment (baijumeswani) [reply orttraining/orttraining/training_api/optimizer.cc:76]: The checkpoint file will always use flatbuffers (no protobuf). But the models can be either ONNX models (in a non minimal build) or an ort format model (no protobuf) for the minimal build.

Comment (baijumeswani) [reply orttraining/orttraining/training_api/optimizer.cc:76]: In short users can either provide an ort format model or an onnx model in a full build.
And only provide an ort format model in a minimal build.
This is similar to the way inference APIs are designed.

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**